Informatics II/001/Task 2/Patertuck Solution/Task2.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+int main() {
+    int dimension = 2;
+    int matrix[dimension*dimension];
+    int matrixsqr[dimension*dimension];
+    int i, j, column, row_start;
+
+    for (i=0; i<dimension * dimension; i++){
+        scanf("%d", &matrix[i]);
+    }
+    //calculating matrixsqr
+    for (i=0; i<dimension * dimension; i++){
+        column = i % dimension;
+        row_start = i - column;
+        matrixsqr[i] = 0;
+
+        for (j=0; j<dimension; j++){
+            matrixsqr[i] += matrix[row_start + j] * matrix[j * dimension + column];
+        }
+    }
+
+    //printing result, could probably be done smoother
+    for (i=0; i<dimension * dimension + 1; i++){
+        if (i % dimension == 0 & i != 0){
+            for (j=0;j<dimension;j++){
+                printf("%d ", matrixsqr[i-dimension+j]);
+            }
+            printf("\n");
+            if (i==dimension*dimension){
+                break;
+            }
```

Comment:
this if is not necessary because with the condition i < dim * dim + 1 the loop already stops at dim * dim

Reply:
Ahh yes i see. In this case, lgtm. In my solution i just run the first loop line by line with a second loop on each line running from left to right 2 * dim times and if the index is under the center print mat a and over mat b.

Reply:
Thanks for the explanation!